huaweicloud/services/aom/resource_huaweicloud_aom_alarm_rule.go

```diff
@@ -0,0 +1,418 @@
+		return diag.Errorf("error creating AOM alarm rule %s: %s", createOpts.AlarmRuleName, err)
+	}
+
+	d.SetId(strconv.FormatInt(*response.AlarmRuleId, 10))
+
+	return resourceAlarmRuleRead(ctx, d, meta)
+}
+
+func resourceAlarmRuleRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.HcAomV2Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("error creating AOM client: %s", err)
+	}
+
+	response, err := client.ShowAlarmRule(&aom.ShowAlarmRuleRequest{AlarmRuleId: d.Id()})
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving AOM alarm rule")
+	}
+
+	allRules := *response.Thresholds
+	if len(allRules) != 1 {
+		return diag.Errorf("error retrieving AOM alarm rule %s", d.Id())
+	}
+	rule := allRules[0]
+	logp.Printf("[DEBUG] Retrieved AOM alarm rule %s: %#v", d.Id(), rule)
+
+	alarm_level, _ := strconv.Atoi(*rule.AlarmLevel)
+
+	mErr := multierror.Append(nil,
+		d.Set("region", config.GetRegion(d)),
+		d.Set("alarm_name", rule.AlarmRuleName),
+		d.Set("alarm_description", rule.AlarmDescription),
+		d.Set("alarm_level", alarm_level),
+		d.Set("metric_name", rule.MetricName),
+		d.Set("alarm_actions", rule.AlarmActions),
+		d.Set("ok_actions", rule.OkActions),
+		d.Set("alarm_enabled", rule.IdTurnOn),
+		d.Set("alarm_action_enabled", rule.ActionEnabled),
+		d.Set("comparison_operator", rule.ComparisonOperator),
+		d.Set("evaluation_periods", rule.EvaluationPeriods),
+		d.Set("insufficient_data_actions", rule.InsufficientDataActions),
+		d.Set("namespace", rule.Namespace),
+		d.Set("period", rule.Period),
+		d.Set("state_value", rule.StateValue),
+		d.Set("state_reason", rule.StateReason),
+		d.Set("statistic", rule.Statistic),
+		d.Set("threshold", rule.Threshold),
+		d.Set("unit", rule.Unit),
+	)
+
+	var dimensions []map[string]interface{}
+	for _, pairObject := range *rule.Dimensions {
+		dimension := make(map[string]interface{})
+		dimension["name"] = pairObject.Name
+		dimension["value"] = pairObject.Value
+
+		dimensions = append(dimensions, dimension)
+	}
+	mErr = multierror.Append(mErr, d.Set("dimensions", dimensions))
+
+	if err := mErr.ErrorOrNil(); err != nil {
+		return fmtp.DiagErrorf("error setting AOM alarm rule fields: %w", err)
+	}
+
+	return nil
+}
+
+func resourceAlarmRuleUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.HcAomV2Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("error creating AOM client: %s", err)
+	}
+
+	updateOpts := aom.UpdateAlarmRuleParam{
+		AlarmRuleName: d.Get("alarm_name").(string),
+	}
+
+	if d.HasChange("alarm_description") {
+		updateOpts.AlarmDescription = utils.String(d.Get("alarm_description").(string))
+	}
+	if d.HasChange("alarm_level") {
+		alarmLevel := d.Get("alarm_level").(int)
+		var alarmLevelToReq aom.UpdateAlarmRuleParamAlarmLevel
```

Comment:
suggest to add a function to convert level to struct

Reply:
fixed